recipe/meta.yaml

```diff
@@ -40,13 +40,13 @@ requirements:
     - libitk-devel
     - libitk
     - nlohmann_json
-    - parallelproj *=cpu*  # [cuda_compiler_version == "None"]
-    - parallelproj *=cuda*  # [cuda_compiler_version != "None"]
+    - parallelproj
     - root_base  # [linux]
     - zlib  # [win]
   run:
     - python
     - {{ pin_compatible('numpy') }}
+    - __cuda  # [cuda_compiler_version != "None"]
```

Comment:
not sure if we need this, but I'm fine with having it. In any case we do need a change, as at present it's installing parallelproj-cpu but stir-cuda!

Reply:
the run requirement on `__cuda` is need to avoid the issue that `mamba` installs the cuda version of stir on on-cuda systems

Reply:
@KrisThielemans at the very end of the [build log of linux with cuda v10](https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/666158/logs/204) it says:
```
2023-03-01T21:48:53.9931392Z     parallelproj:                         1.2.14-cuda11.1h12f7317_1  conda-forge
```
So at least in this build, the correct cuda parallelproj variant gets installed.

@KrisThielemans In which log file did you observer that `parallelproj:cpu` was installed on a `cuda` system?

Reply:
ok, I see in the cuda 11 build it uses the `parallelproj:cpu` :-(
-> would have been to easy / logical if it worked
-> good news: we can easily fix that by using the cuda_compiler selectors in the recipe

Reply:
`parallelproj` is also still missing the `run_exports` (to make sure that if we build against `parallelproj:cuda`, we also need `parallelproj:cuda` at runtime.
So please wait before merging this **PR**